components/layout/layout_task.rs

```diff
@@ -117,6 +119,9 @@ pub struct LayoutTask {
     /// The ID of the pipeline that we belong to.
     pub id: PipelineId,
 
+    /// The URL of the pipeline that we belong to.
+    pub url: Url,
```

Comment:
As we discussed on IRC: there's no SharedLayoutContext readily available from the layout task's event loop, which is where it's needed. A SharedLayoutContext does get built during reflow, but it's built from data passed in via the `Reflow` message.